### ISSUE TITLE
state sync failures observed with atomic root mismatch

### ISSUE THREAD

**Issue by darioush** [ava-labs/coreth#280]
**Describe the bug**
We have received 2 instances of reports with logs such as:

`FATAL <C Chain> handler/handler.go:335 shutting down chain {"reason": "received an unexpected error", "error": "failed to notify VM that bootstrapping has started: synced root (0x35ae5c47aeab8d06a088b70034833af6414dfdb228f1f6e2a3cf65e94f4f04e5) does not match expected (0x48d9b80aa254dad0cbd11c36414eb36af46a595bbad975fefe5462a70eea9bd6) for atomic trie while processing async message: `

**To Reproduce**
Unknown.

**Expected behavior**
No error during state sync.

**Comment by michaelbu-zerohash** [comment]
Attached C.log
[clogs.tar.gz](https://github.com/ava-labs/coreth/files/12010070/clogs.tar.gz)

**Comment by michaelbu-zerohash** [comment]
[C-2023-07-09T01-46-58.180.log](https://github.com/ava-labs/coreth/files/12010139/C-2023-07-09T01-46-58.180.log)
[C-2023-07-09T09-42-37.460.log](https://github.com/ava-labs/coreth/files/12010140/C-2023-07-09T09-42-37.460.log)
[C-2023-07-09T17-32-05.627.log](https://github.com/ava-labs/coreth/files/12010141/C-2023-07-09T17-32-05.627.log)

**Comment by michaelbu-zerohash** [comment]
[C-2023-07-07T14-56-09.843.log](https://github.com/ava-labs/coreth/files/12010147/C-2023-07-07T14-56-09.843.log)
[C-2023-07-07T23-30-45.430.log](https://github.com/ava-labs/coreth/files/12010148/C-2023-07-07T23-30-45.430.log)
[C-2023-07-08T08-10-21.166.log](https://github.com/ava-labs/coreth/files/12010149/C-2023-07-08T08-10-21.166.log)
[C-2023-07-08T16-45-23.689.log](https://github.com/ava-labs/coreth/files/12010150/C-2023-07-08T16-45-23.689.log)

**Comment by ceyonur** [comment]
@darioush Could this be already fixed?

**Comment by darioush** [comment]
I don't think we found any reason that this would be fixed. Although I haven't seen any reproduction for it either.
